pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/predicate/StepPredicatePair.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.predicate;
+
+import com.datastax.oss.pulsar.functions.transforms.TransformContext;
+import com.datastax.oss.pulsar.functions.transforms.TransformStep;
+import java.util.function.Predicate;
+import lombok.Data;
+
+@Data
+public class StepPredicatePair {
+  private final TransformStep transformStep;
+  private final Predicate<TransformContext> predicate;
```

Comment:
This is very interesting.

It is not obvious if the predicate is evaluated on the initial message or on the intermediate record.

I think that current implementation, based on the current status of the Record is good but we have to document it clearly

Reply:
Agreed that the predicate should apply to the current status.

Reply:
Makes sense. I'm not sure if it clarifies, but I can move the `predicate` to the `TransformStep` class so that it is more encapsulated in the step and since the step operates on the current state, so will the predicate.  

The only reason I kept them separate (and grouped in pairs) is because the step actually has nothing to do with the predicate, it is the higher level processor that takes a "process, don't process" decision. Added predicate evaluation function on something like an "AbstractTransformStep" sounded like an overkill... Will add the java docs for now.